brat/Brat/Checker/Monad.hs

```diff
@@ -213,20 +206,17 @@ handler :: Free CheckingSig v
         -> Context
         -> Graph
         -> Namespace
-        -> Either Error (v,Context,([TypedHole],Graph),Namespace)
-handler (Ret v) ctx g ns = return (v, ctx, ([], g), ns)
+        -> Either Error (v,Context,([TypedHole],Graph))
+handler (Ret v) ctx g _ = return (v, ctx, ([], g))
```

Comment:
I am even wondering if we could *remove* the `Namespace` parameter to handler. Have the top-level command(s) (i.e. `GetNS`+`SetNS` or `SplitNS`) die with an error, and rely on running a `localNS` inside it. You could make `run` still take a Namespace and invoke `handler (localNS c)`.... (so, a bit like `localFC`)

Reply:
I was thinking the same thing!